core/services/sleeper_task.go

```diff
@@ -62,6 +70,7 @@ func (s *sleeperTask) workerLoop() {
 		select {
 		case <-s.waker:
 			s.worker.Work()
+			s.workerWG.Done()
```

Comment:
I'd probably `defer` the `.Done()` call and move it above `.Work()` so that it'll still be called if the worker panics.  Otherwise, it looks like `.Stop()` could hang indefinitely.

Reply:
Good one @spooktheducks 🙏. Updated